connector/src/main/scala/com/microsoft/kusto/spark/utils/KeyVaultUtils.scala

```diff
@@ -15,67 +16,72 @@ object KeyVaultUtils {
   val StorageAccountId = "blobStorageAccountName"
   val StorageAccountKey = "blobStorageAccountKey"
   val Container = "blobContainer"
-  var cachedClient: KeyVaultClient = _
+  var cachedClient: SecretClient = _
 
-  private def getClient(clientID: String, clientPassword: String): KeyVaultClient ={
+  private def getClient(uri: String, clientID: String, clientPassword: String, authority: String): SecretClient ={
     if(cachedClient == null) {
-      cachedClient = new KeyVaultADALAuthenticator(clientID, clientPassword).getAuthenticatedClient
+      cachedClient = new KeyVaultADALAuthenticator(uri, clientID, clientPassword, authority).getAuthenticatedClient
     }
     cachedClient
   }
 
-  @throws[CloudException]
   @throws[IOException]
   def getStorageParamsFromKeyVault(keyVaultAuthentication: KeyVaultAuthentication): KustoStorageParameters = {
     keyVaultAuthentication match {
       case app: KeyVaultAppAuthentication =>
-        val client = getClient(app.keyVaultAppID, app.keyVaultAppKey)
+        val client = getClient(app.uri, app.keyVaultAppID, app.keyVaultAppKey, app.authority)
         getStorageParamsFromKeyVaultImpl(client, app.uri)
       case certificate: KeyVaultCertificateAuthentication => throw new UnsupportedOperationException("certificates are not yet supported")
     }
   }
 
-  @throws[CloudException]
   @throws[IOException]
   def getAadAppParametersFromKeyVault(keyVaultAuthentication: KeyVaultAuthentication): AadApplicationAuthentication={
     keyVaultAuthentication match {
       case app: KeyVaultAppAuthentication =>
-        val client = getClient(app.keyVaultAppID, app.keyVaultAppKey)
+        val client = getClient(app.uri, app.keyVaultAppID, app.keyVaultAppKey, app.authority)
         getAadAppParamsFromKeyVaultImpl(client, app.uri)
-      case certificate: KeyVaultCertificateAuthentication => throw new UnsupportedOperationException("certificates are not yet supported")
+      case _: KeyVaultCertificateAuthentication => throw new UnsupportedOperationException("certificates are not yet supported")
```

Comment:
I'm new to scala, but don't we want to match case of "certificate" if we're going to throw an exception about certificates not being supported?